### DISCUSSION TITLE
is there a way to migrate the share screen video to the static UI?

### DISCUSSION THREAD

**Discussion by lukewhitingdev** [mozilla/hubs#2935] [General Questions & Discussion] [answered]
Just wondering as I wanted to implement a system where a admin could view others screens and teleport to them for help if needed. Im a bit new to WebRTC so was trying to see if i could interface with the one built into hubs (that powers the existing share screen function) rather remaking the wheel.

**Comment by emclaren** [comment, marked as answer]
Hi Luke, if you are using Hubs Cloud, I don't see any reason why you wouldn't be able to make a fork of the Hubs Client and modify the code so that when someone shares their screen it could load somehow in your UI instead of in the 3D space. 

Currently we have a feature that allows you to pin screens to the bottom of your viewport - perhaps some of the code used there could be useful: https://github.com/mozilla/hubs/pull/2081 . I believe that users would need to opt to share their screen with you (due to privacy settings). 

We are hoping to add a feature soon that would let you teleport to different users in the room, however this work is not yet underway. You can add feedback and track our progress on this here: https://github.com/mozilla/hubs/issues/2585

**Reply by netpro2k** [reply]
As mentioned, you should be able to do this with just some client changes. Here is a pointer to where you might start to wire something like this in https://github.com/mozilla/hubs/blob/master/src/scene-entry-manager.js#L400 .. In that function the local media stream is set up and set to broadcast through our webrtc backend (via `connection.adapter.setLocalMediaStream`). You would then need to handle receiving other people's streams as well, but ultimately they end up as `<video>` tags, so you absolutely should be able to land them in a 2D UI. 

Though this is not in our direct roadmap, it is something we have talked about a bunch so it would be really great to see!
